### ISSUE TITLE
average returns wrong value for nullable field

### ISSUE THREAD

**Issue by zaki50** [realm/realm-java#1803]
Reported by @dev75py on twitter. https://twitter.com/dev75py/status/667469537986527232

On 0.85.0, `RealmQuery.average()` returns wrong value for nullable field.

```
    public void testAverage() {
        final NullTypes obj1 = new NullTypes();
        obj1.setId(1);
        obj1.setFieldIntegerNotNull(3);
        obj1.setFieldIntegerNull(3);

        final NullTypes obj2 = new NullTypes();
        obj2.setId(2);
        obj2.setFieldIntegerNotNull(4);
        obj2.setFieldIntegerNull(4);

        testRealm.beginTransaction();
        testRealm.where(NullTypes.class).findAll().clear();
        testRealm.copyToRealm(obj1);
        testRealm.copyToRealm(obj2);
        testRealm.commitTransaction();

        assertEquals(3.5, testRealm.where(NullTypes.class).average(NullTypes.FIELD_INTEGER_NOT_NULL));
        assertEquals(3.5, testRealm.where(NullTypes.class).average(NullTypes.FIELD_INTEGER_NULL));
    }
```

The last assertion fails with actual value `9.0`.

**Comment by cmelchior** [comment]
@zaki50 Hasn't this been fixed?

**Comment by zaki50** [comment]
@cmelchior I think it's fixed.
I'll confirm and close it

**Comment by zaki50** [comment]
This issue was fixed in 0.86.0
